formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/JsonLexer.kt

```diff
@@ -131,6 +131,32 @@ internal class ReaderJsonLexer(
         preload(spaceLeft)
     }
 
+    override fun consumeStringChunked(consumeChunk: (stringChunk: String) -> Unit) {
+        consumeNextToken(STRING)
+        var currentPosition = this.currentPosition
+        var lastPosition = currentPosition
+        var char = source[currentPosition] // Avoid two range checks visible in the profiler
+        while (char != STRING) {
+            if (++currentPosition >= source.length) {
+                // end of chunk
+                writeRange(lastPosition, currentPosition, consumeChunk)
+                currentPosition = prefetchOrEof(currentPosition)
+                if (currentPosition == -1)
+                    fail("EOF", currentPosition)
+                lastPosition = currentPosition
+            }
+            char = source[currentPosition]
+        }
+        writeRange(lastPosition, currentPosition, consumeChunk)
+        this.currentPosition = currentPosition + 1 // Consume closing '"' (STRING)
+    }
+
+    private fun writeRange(fromIndex: Int, toIndex: Int, consumeChunk: (stringChunk: String) -> Unit) {
+        val tmp = StringBuilder()
```

Comment:
Creating a StringBuilder every time is not a fast operation (especially since it will reallocate a large amount of memory to accommodate larger ranges). You can re-use like it's done with `escapedString`, or better, just use `substring` method of ReaderJsonLexer

Reply:
fixed